CONTRIBUTING.md

```diff
@@ -82,12 +84,45 @@ The following tools are required:
 - [git](https://git-scm.com)
 - [python](https://www.python.org) (v3.8+)
 - [pip](https://pypi.org/project/pip/) (v23.0+)
+- [poetry](https://python-poetry.org/docs/#installation) (v1.8.3+)
+  - Poetry should always be installed in a dedicated virtual environment to isolate it from the rest of your system. It should in no case be installed in the environment of the project that is to be managed by Poetry. This ensures that Poetry’s own dependencies will not be accidentally upgraded or uninstalled.
+- [tox](https://tox.wiki/en/4.15.1/installation.html) (v4.15.1+)
+  - Just like `poetry` install `tox` in an isolated virtual environment
 
 Installation:
-``` 
-pip install -U datasets
-pip install -e .
+
+```bash 
+: Install poetry and tox in an isolated virtual environment
+python3 -m venv isolated
```

Comment:
but why we care so much about `poetry` being updated? will there really be some random package to update peotry? If so then we should pay the same care to `tox` but when dont..

Can we add `poetry` as a dep in the `tox` environment, and have tox manage it?

Reply:
Poetry depends on `builds` package which conflicts with our local `builds` package.

tox's deps are also installed in same venv.

Reply:
oic.. guess `builds` is too common of a package name.